hookdocs/useful-snippets.md

```diff
@@ -185,6 +172,87 @@ add_filter(
 );
 ```
 
+5. **Add required provider fields to the settings**
+ClassifAI uses a React-based settings panel. To add the necessary provider fields, include the following code in a JavaScript file, and save it within your plugin's directory.
+
+_**Note:** The provided code snippet uses modern JavaScript (ESNext and JSX), which requires a build step to compile it into a browser-compatible format. We recommend using the [@wordpress/scripts](https://www.npmjs.com/package/@wordpress/scripts) package to manage the build process. For a step-by-step guide on how to install and use this package, check out this tutorial: [Get started with wp-scripts](https://developer.wordpress.org/block-editor/getting-started/devenv/get-started-with-wp-scripts/)._
+
+```js
+/**
+ * WordPress dependencies
+ */
+import { registerPlugin } from '@wordpress/plugins';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Fill, TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MyProviderSettings = () => {
+	const providerName = 'my_provider';
+	const featureSettings = useSelect( ( select ) =>
+		select( 'classifai-settings' ).getFeatureSettings()
+	);
+	const { setFeatureSettings } = useDispatch( 'classifai-settings' );
+	const providerSettings = featureSettings[ providerName ] || {};
+	const setProviderSettings = ( data ) =>
+		setFeatureSettings( {
+			[ providerName ]: {
+				...providerSettings,
+				...data,
+			},
+		} );
+
+	return (
+		<Fill name="ClassifAIProviderSettings">
```

Comment:
**Not to self:**

Find out a way to make this easier so that 3rd-party devs don't have to write `Fill`.
Maybe try exporting a component on the `window` object which can be used to inject custom Provider settings.